internal/controlplane/handlers_profile.go

```diff
@@ -612,7 +654,7 @@ func (s *Server) UpdateProfile(ctx context.Context,
 		return nil, status.Errorf(codes.Internal, "failed to get profile: %s", err)
 	}
 
-	oldRules, err := s.getRulesFromProfile(ctx, oldProfile, entityCtx)
+	oldRules, err := s.getRulesFromOldProfile(ctx, oldProfile, entityCtx)
```

Comment:
`getRulesFromProfile` is a generic method to get rules from a profile. It nowhere mentions whether the profile has been populated with default rule names or not. In case, it is not, then the method currently doesn't use `computeRuleName(..)` , so the returned rules won't `name` populated, which isn't ideal. Also, for old profiles, we don't want to `computeRuleName(..)`, hence the name change.

Also, this method is renamed back to `getRulesFromProfile` in the backfilling PR, when we are sure every retrieved profile from db will have a name.